content/reference/stacks.md

```diff
@@ -0,0 +1,277 @@
+* **Base Components**: For any configuration that you want to share across all instances of a component.
+  * For example, you're using the `eks` component and you want to ensure all of you EKS clusters are using the same Kubernetes version, you create a `stacks/catalogs/eks-globals.yaml` file which specifies the `eks` component's `vars.kubernetes_version`. You can then `import` that base component configuration in any stack file that uses the `eks` component.
+  * More information in the below section.
+
+## Component Inheritance
+
+Using component catalogs, you can define default values for all instances of a component across your stacks. But it is also important to note that you can provide default values for multiple instances of a component in a single stack using the component inheritance pattern via the `component` key / value:
+
+```yaml
+# stacks/catalogs/s3-globals.yaml
+components:
+  terraform:
+    s3-bucket:
+      vars:
+        enabled: false
+        user_enabled: false
+        acl: private
+        grants: null
+        versioning_enabled: true
+```
+
+```yaml
+# stacks/uw2-dev.yaml
+imports:
+  - catalogs/s3-globals
+  - catalogs/dev-globals
+  - catalogs/uw2-globals
+
+components:
+  terraform:
+    public-images:
+      component: s3-bucket
+      vars:
+        enabled: true
+        acl: public
+        name: public-images
+
+    export-data:
+      component: s3-bucket
+      vars:
+        enabled: true
+        name: export-data
+
+    # ...
+```
+
+In the above, we're able to utilize the default settings provided via the `s3-bucket` base component catalog, while also creating multiple instances of the same component and providing our own overrides. This enables maximum reuse of global component configuration.
+
+## Terraform Workspace Names
+
+`atmos` and accompanying terraform automation modules like [terraform-spacelift-cloud-infrastructure-automation](https://github.com/cloudposse/terraform-spacelift-cloud-infrastructure-automation) will automatically create terraform [workspaces](https://www.terraform.io/docs/language/state/workspaces.html) when managing components. These workspaces derive their names from the stack name and the component name in question following this pattern: `$env-$stage-$component`. The result is workspace names like `ue2-dev-eks` or `uw2-prod-mq-broker`.
+
+# Stack Schema
+
+<!--
+TODO: Is the vault backend supported as of today?
+TODO: Should we mention the `remote` backend considering the current state of the TFC automation module?
+-->
+
+[The official JSON Schema document for Stacks can be found here](https://github.com/cloudposse/atmos/blob/master/docs/schema/stack-config-schema.json). The below is a walk through of a complete example utilizing all capabilities.
+
+```yaml
+# stacks/ue2-dev.yaml
+s3-bucket:
+      vars:
```

Comment:
whitespace

Reply:
structure wrong

Reply:
Will fix.

Reply:
Fixed. Removed this completely since I believe it was a bad copy / pasta.